### PR TITLE
Don't use "coins" and "tokens" anymore, add comments 

### PR DESCRIPTION
**Pull request by dpc**
Rename to `Note`s and `Nonce`s.

Discussion notes in https://github.com/fedimint/fedimint/issues/501

On top of #492 to spare me fixing rebase conflicts.

### CODE REVIEW

File: client/cli/src/main.rs [hunk 1]
```diff
@@ -42,13 +42,13 @@ enum Command {
     /// Reissue tokens received from a third party to avoid double spends
     Reissue {
         #[clap(parse(from_str = parse_coins))]
-        coins: Coins<SpendableCoin>,
+        coins: TieredMulti<SpendableNote>,
```
Comment (jkitman) [inline client/cli/src/main.rs:45]: What about `TieredVec`?  `Multi` is a bit ambiguous.

Guess the parameter names should be changed, were you going to do that as well?

Comment (dpc) [reply client/cli/src/main.rs:45]: I took the `TieredMulti` from the existing `TieredMultiZip`. I'm not too happy about this name and slightly prefer `TieredVec` as well. But it's also not ideal. :D

Comment (jkitman) [reply client/cli/src/main.rs:45]: Another option is `Denomination` which is even more clear in the context of bank `Note`, but maybe too long to type...

Will you also change the parameter name e.g. `coins:` to something like `notes` or `ecash`?

Comment (dpc) [reply client/cli/src/main.rs:45]: I have a follow up PR with more renames, etc.

This enum variant could be just a newtype and there would be no need to name anything.

**Review by justinmoon — APPROVED** [review]
LGTM. `Note` is better than `Coin` and `Token` IMO. Can always rename again in the future if we think of a better name :)

**Comment by justinmoon** [comment]
BTW I attempted "auto-merge" here and I guess it failed?

**Comment by justinmoon** [comment]
Now we have this second commit 
image

**Comment by dpc** [comment]
Merge conflict of a kind. Force pushed a fix now.

**Review by justinmoon — APPROVED** [review]

**Pull request merged**